backend/src/sms/sms.service.ts

```diff
@@ -25,7 +36,10 @@ export class SmsService {
         const attemptMessage = await this.sendMessageToUser(messageRequest, settings.endPoint);
         if (attemptMessage) {
             return true;
-        } else {
+        } else if(smsSettingsError){
```

Comment:
Im not sure if this code-block is reachable, since you already thow error at the start of the function?

Reply:
Yeah this shouldn't be here. Actually the if on line 22 should be just a "if (!settings)" and on line 26 the else if should be just an if (since they are not related) like this: "if (smsSettingsError)". Then the else if on line 39 could be removed. Or am I missing something?

Reply:
@ohel about line 39 that was my mistake. pushing this fix right away

but about the first part, what i was trying:
line 20: get stack trace to check if the call to `sendVerificationSMS()` is coming from `registerJunior()`
line 22: `if(!settings && smsSettingsError)`-  !settings checks if telia configs are missing && smsSettingsError checks if the call to the function came from registerJunior() => and display an error saying "SMS is not working, but user created'
line26: `else if (!settings)` - checking if the telia configs are missing for literally every other case other than registerJunior() where an SMS should be sent.

so this should stay the same

Reply:
Because in any case settings should exist - it's an error and _nothing_ works if they don't - it should be the first check.

Also, does reading a possible error stack really work like that, just construct a new Error and read its stack? :o What happens if there's no error, is the indexing [2] safe?